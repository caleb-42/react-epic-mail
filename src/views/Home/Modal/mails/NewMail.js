import React from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from 'prop-types';
import TextInput from '@components/Inputs/TextInput';
import RadioInput from '@components/Inputs/RadioInput';
import SubmitBtn from '@components/ActionBtn';
import callToast from '@components/Toast';
import * as mailActions from "@redux/actions/mailActions";
import { addMailInputs as inputs } from './inputs';


// eslint-disable-next-line import/prefer-default-export
export const NewMailModal = ({ response, actions }) => {

  const [values, setValues] = React.useState({
    msgTo: 'stranger'
  });

  const [sendSaving, setSendSaving] = React.useState(false);
  const [draftSaving, setDraftSaving] = React.useState(false);

  const handleSubmit = async (event, save, thunk) => {
    event.preventDefault();
    save(true);
    const newValues = (({ msgTo, ...rest }) => rest)(values);
    console.log(newValues);
    thunk(newValues, () => {
      save(false);
    });
  }

  const handleChange = e => {
    e.persist();
    setValues(prevState => ({ ...prevState, [e.target.name]: e.target.value }));
  }

  return (
    <div className="new-mail">
      <header>
        <h2>New mail</h2>
      </header>
      <form className="form-hd" onSubmit={handleSubmit} autoComplete="off">
        <div className="contactOptions lg:my-12 my-8">
          {
            inputs.radio.map((inp, index) => (
              <RadioInput
                {...inp}
                key={index}
                checked={values[inp.name] === inp.value}
                name={inp.name}
                onChange={handleChange}
              />
            ))
          }
        </div>
        <div className="lg:w-3/4 w-4/5 mx-auto">
          {
            inputs.text.map((inp, index) => (
              <TextInput
                {...inp}
                key={index}
                onChange={handleChange} />
            ))
          }
        </div>
        <div className="text-center smsSend">
          <input type="checkbox" name="sendsms" onChange={handleChange} value={true} id="sms" />
          <label htmlFor="sms" className="f-15 ml-2">Send SMS</label>
        </div>
        <div className="action">
          <span>
            <SubmitBtn
              className="sendBtn"
              title="send"
              disabled={draftSaving || sendSaving}
              onClick={e => handleSubmit(e, setSendSaving, actions.sendMail)}
              saving={sendSaving}
            />
          </span>
          <span>
            <SubmitBtn
              className="sendBtn"
              title="save"
              disabled={draftSaving || sendSaving}
              onClick={e => handleSubmit(e, setDraftSaving, actions.saveMail)}
              saving={draftSaving}
            />
          </span>
        </div>
      </form>
    </div >

  );
};

NewMailModal.propTypes = {
  handleChange: PropTypes.func
};

function mapStateToProps(state) {
  return {
    response: state.mails.response
  }
}

const matchDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(mailActions, dispatch)
})

export default connect(mapStateToProps, matchDispatchToProps)(NewMailModal);