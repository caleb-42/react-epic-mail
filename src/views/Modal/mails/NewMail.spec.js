import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore from "@redux/configureStore";
import NewMail from "./NewMail";

const store = configureStore();

let defaultProps;
const FakeNewMail = (args) => {
  defaultProps = {
    response: { menu: 'Mails' },
    actions: {
      sendMail: jest.fn(() => { }),
      saveMail: (jest.fn(() => { }))
    }
  }
  const props = { ...defaultProps, ...args };
  const store = configureStore(props);
  return render(
    <ReduxProvider store={store}>
      <NewMail />
    </ReduxProvider>
  )
}

describe('SNAPSHOT TEST', () => {

  it("Header Component", () => {
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <NewMail />
      </ReduxProvider >
    );
    expect(tree).toMatchSnapshot();
  });
})
