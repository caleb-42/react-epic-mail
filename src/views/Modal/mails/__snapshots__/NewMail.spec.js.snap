// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT TEST Header Component 1`] = `
<div
  className="new-mail"
>
  <header>
    <h2>
      New mail
    </h2>
  </header>
  <form
    autoComplete="off"
    className="form-hd"
  >
    <div
      className="contactOptions lg:my-12 my-8"
    >
      <span
        className="input-group"
      >
        <input
          checked={true}
          id="stranger"
          name="msgTo"
          onChange={[Function]}
          type="radio"
          value="stranger"
        />
        <label
          className=""
          htmlFor="stranger"
        >
          Stranger
        </label>
      </span>
      <span
        className="input-group"
      >
        <input
          checked={false}
          id="contact"
          name="msgTo"
          onChange={[Function]}
          type="radio"
          value="contact"
        />
        <label
          className=""
          htmlFor="contact"
        >
          Contact
        </label>
      </span>
      <span
        className="input-group"
      >
        <input
          checked={false}
          id="group"
          name="msgTo"
          onChange={[Function]}
          type="radio"
          value="group"
        />
        <label
          className=""
          htmlFor="group"
        >
          Group
        </label>
      </span>
    </div>
    <div
      className="lg:w-3/4 w-4/5 mx-auto"
    >
      <div
        className="flex flex-col input-group items-center"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="johndoe@epicmail.com"
          required={true}
          type="email"
        />
      </div>
      <div
        className="flex flex-col input-group items-center"
      >
        <label
          htmlFor="subject"
        >
          Subject
        </label>
        <input
          className="input"
          name="subject"
          onChange={[Function]}
          placeholder="Subject"
          required={true}
          type="text"
        />
      </div>
      <div
        className="flex flex-col input-group items-center"
      >
        <label
          htmlFor="message"
        >
          Message
        </label>
        <textarea
          className="input"
          name="message"
          onChange={[Function]}
          placeholder="Message"
          required={true}
          rows={5}
        />
      </div>
    </div>
    <div
      className="text-center smsSend"
    >
      <input
        id="sms"
        name="sendsms"
        onChange={[Function]}
        type="checkbox"
        value={true}
      />
      <label
        className="f-15 ml-2"
        htmlFor="sms"
      >
        Send SMS
      </label>
    </div>
    <div
      className="action"
    >
      <span>
        <button
          className="sendBtn"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          send
        </button>
      </span>
      <span>
        <button
          className="sendBtn"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          save
        </button>
      </span>
    </div>
  </form>
</div>
`;
